src/main/kotlin/no/nav/syfo/melding/MeldingService.kt

```diff
@@ -79,4 +78,14 @@ class MeldingService(
             ?.behandlerRef
             ?: throw IllegalStateException("Fant ikke behandlerRef for samtale $conversationRef, kunne ikke knyttes til melding fra behandler")
     }
+
+    fun getUnjournalforteMeldinger(): List<Pair<PMelding, ByteArray>> { // TODO: Finne bedre navn?
+        return database.connection.getDialogmeldingerToJournalfore().map { (pMelding, pPdf) ->
+            Pair(pMelding, pPdf.pdf)
+        }
+    }
+
+    fun updateDialogmeldingJournalpostId(dialogmelding: PMelding, journalpostId: String) {
+        database.connection.updateMeldingJournalpostId(dialogmelding, journalpostId)
```

Comment:
Oj, dette er krise: lekker en connection. Må bruke connection.use { ... }

Reply:
Er det ikke bra nok å gjøre `this.use{}` inne i query-metoden? 🤔

Reply:
Kunne ikke alle datbasekallene her vært extension functions på DatabaseInterface så kan database-laget (query-metodene) ta seg av connection-håndteringen? Synes det er fint om servicen slipper å håndtere connections (med mindre det trengs om man skal gjøre flere ting i én connection før man commiter f.eks.)

Reply:
Ja, det vil jo funke slik dere har gjort det her, men det Anders foreslår er ryddigere.